scripts/build.py

```diff
@@ -0,0 +1,213 @@
+    """
+    triplet_name = [arch]
+    if is_windows_os():
+        triplet_name.append('windows')
+        # For windows, default is to build dynamic. Hence we specify static.
+        # For mac/linux, default is to build static libraries
+        triplet_name.append('static')
+    elif is_mac_os():
+        triplet_name.append('osx')
+    elif is_linux_os():
+        triplet_name.append('linux')
+    else:
+        raise ValueError('Unsupported operating system \
+                          We currently support windows, mac, linux')
+
+    return '-'.join(triplet_name)
+
+
+def step_cmake_configure(args):
+    """ CMake configure
+
+    If you are seeing problems when running this multiple times,
+    make sure to clean/delete previous build directory
+
+    Args:
+        args (`argparse.Namespace`): Parsed command line arguments
+    """   
+    cmd = ['cmake', '-S', '.', '-B', args.build_dir]
+    
+    # If generator is not specifed, default for platform is used by cmake, else 
+    # use the specified value
+    if args.generator:
+        cmd.extend(['-G', args.generator])
+    if args.config:
+        cmd.append('-DCMAKE_BUILD_TYPE={0}'.format(args.config))
+    if args.build_tests:
+        cmd.append('-DFIREBASE_CPP_BUILD_TESTS=ON')
+        cmd.append('-DFIREBASE_FORCE_FAKE_SECURE_STORAGE=ON')
+    # If you want to avoid vcpkg altogether and have cpp dependencies installed elsewhere
+    if not args.no_vcpkg:
+        vcpkg_toolchain_file_path = os.path.join(REPO_ROOT_DIR, 'external', 'vcpkg', 
+                                                'scripts', 'buildsystems', 'vcpkg.cmake')
+        
+        cmd.append('-DCMAKE_TOOLCHAIN_FILE={0}'.format(vcpkg_toolchain_file_path))
+        cmd.append('-DVCPKG_TARGET_TRIPLET={0}'.format(get_vcpkg_triplet(args.arch)))
+    
+    cmd.append('-DFIREBASE_INCLUDE_FIRESTORE=OFF')
+    run_command(cmd, cwd=REPO_ROOT_DIR)
+
+
+def step_cmake_build(args):
+    """ CMake build
+
+    Args:
+        args (`argparse.Namespace`): Parsed command line arguments
+    """   
+    cmd = ['cmake', '--build', args.build_dir, '-j', str(args.j)]
+    if args.target:
+        targets = ['--target']
+        targets.extend(args.target)
+        cmd.extend(targets)
+    run_command(cmd, cwd=REPO_ROOT_DIR)
+
+
+def main():
+    args = parse_cmdline_args()
+
+    step_cmake_configure(args)
+    step_cmake_build(args)
+
+
+def parse_cmdline_args():
+    parser = argparse.ArgumentParser(description='Install Prerequisites for building cpp sdk')
+    parser.add_argument('-A', '--arch', default='x64', help='Platform architecture (x64, x86)')
+    parser.add_argument('-G', '--generator', help='Build generator to use')
+    parser.add_argument('--build_dir', default='build', help='Output build directory')
+    parser.add_argument('--build_tests', action='store_true', help='Build unit tests too')
+    parser.add_argument('--no_vcpkg', action='store_true', help='Do not use vcpkg toolchain')
```

Comment:
Is it just the toolchain or all of the other dependencies, too?  Maybe this be something more akin to why it would be used and not expiclity name vcpkg.  Something like --custom_tool_config or something.,

Reply:
So this option was to avoid using any of vcpkg stuff and have users just build without it. This assumes users used some other package manager to install cpp dependencies and everything is accessible by the build system.

Reply:
Again, I think this script's primary purpose is to install prereqs on GHA runners and build on those runners. I don't think it should support every use case. If a developer wants to NOT use vcpkg they can go ahead and set up the prereqs themselves rather than running our own prereqs script.